### PR TITLE
Project settings file

### PR DESCRIPTION
**Pull request by andre2007**
Solves issue https://github.com/dlang/dub/issues/1739

Dub settings file can now also be added to project root folder.
As name ```dub.settings.json``` was used while the file is called ```settings.json``` in other locations (https://dub.pm/settings.html).
I am not 100% sure about the name, in project folder ```settings.json``` might be to generic?

### CODE REVIEW

**Comment by ZombineDev** [comment]
+1 for `dub.settings.json`, or perhaps `dub.global-settings.json` to make it clear that this file is not package specific (like `dub(.selections)?.json` / `dub.sdl`), but overrides the global dub settings.

File: changelog/project-settings.file.dd [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Support of project settings file
```
Comment (ZombineDev) [inline changelog/project-settings.file.dd]:
```suggestion
Support for dub global settings file at the root package level 
```

**Comment by andre2007** [comment]
@ZombineDev thanks for feedback. Both alternatives looks good.

**Comment by andre2007** [comment]
semaphoreci fails with unrelated errors:
- no errors for dmd are shown
- ```/tmp/ccRaYTk4.o: In function `__lambda9':
/home/runner/dub/source/dub/internal/vibecompat/core/log.d:77: undefined reference to `_D3std5stdio24__T10makeGlobalS6stderrZ10makeGlobalFNbNcNdNiZS3std5stdio4File'``` shown for gdc

**Comment by ZombineDev** [comment]
Yep, I think SemaphoreCI stops as soon as one of the elements of the test matrix fails. I'd recommend disabling gdc for now.

**Comment by ZombineDev** [comment]
Actually, I think we should have more success with GDC for GCC 9. Can you change this line (and as needed the ones below) to gdc 9:
https://github.com/dlang/dub/blob/0968f6481d765014f0ac3bc716bca7c27d1e9155/semaphore-ci.sh#L25

File: changelog/project-settings.file.dd (hunk 1 above)
Comment (andre2007) [reply changelog/project-settings.file.dd]: done

**Comment by andre2007** [comment]
@ZombineDev switched to gdc-9

**Comment by andre2007** [comment]
...gdc is now working... but a lot more unrelated test failures ...

**Comment by andre2007** [comment]
Reopening to retrigger tests

**Comment by andre2007** [comment]
@ZombineDev I do not know how to proceed here. The failing GDC tests are completely unrelated.

**Comment by ZombineDev** [comment]
Yes, it's totally unrelated, but unfortunately it looks like that no future PRs could be merged unless we either disable GDC or fix the problems.
Perhaps running a docker image with approximately the same environment will help debug the problems with GDC.

**Comment by ZombineDev** [comment]
Alternatively, I'd be fine with changing line 7 in `semaphore-ci.sh` to `exit 0` and opening a new issue to investigate and fix the problems with GDC.

**Comment by andre2007** [comment]
@ZombineDev thanks, I created issue https://github.com/dlang/dub/issues/1744

**Comment by ZombineDev** [comment]
@andre2007 thanks for your work and persistence as well!

**Pull request merged**